pipeline/inputs/process-exporter-metrics.md

```diff
@@ -0,0 +1,88 @@
+---
+description: >-
+  A plugin based on Process Exporter to collect process level of metrics of system
+  metrics
+---
+
+# Process Exporter Metrics
+
+[Prometheus Node Exporter](https://github.com/prometheus/node_exporter) is a popular way to collect system level metrics from operating systems, such as CPU / Disk / Network / Process statistics.
+Fluent Bit 2.2 onwards includes a process exporter plugin that builds off the Prometheus design to collect process level metrics without having to manage two separate processes or agents.
+
+The Process Exporter Metrics plugin implements collecting of the various metrics available from [the 3rd party implementation of Prometheus Process Exporter](https://github.com/ncabatoff/process-exporter) and these will be expanded over time as needed.
+
+**Important note:** Metrics collected with Process Exporter Metrics flow through a separate pipeline from logs and current filters do not operate on top of metrics.
+
+This plugin is only supported on Linux based operating systems due to heavily relying on proc filesystem.
+
+macOS does not have procfs. Therefore, this plugin won't work for it.
+
+
+## Configuration
+
+| Key                       | Description                                                                            | Default   |
+| ------------------------- | -------------------------------------------------------------------------------------- | --------- |
+| scrape_interval           | The rate at which metrics are collected.                 | 5 seconds |
+| path.procfs               | The mount point used to collect process information and metrics. Read-only is enough   | /proc/    |
+| process\_include\_pattern | regex to determine which names of processes are included in the metrics produced by this plugin | It is applied for all process unless explicitly set. Default is `.+`. |
```

Comment:
Can we clarify the regex handling?

It would be useful to have examples for how to monitor specific processes.
Is it the binary name itself as reported by `ps` for example or something else (e.g. if symlinked with a different name?).
Can we/how do we filter processes with the same name but different args? 
I presume we cannot filter by pid either?

Reply:
1. Yes. The binary names are handled as ps reported names which is limited 15 characters.
2. No. We cannot filtered by command line arguments like as prometheus process_exporter does.
3. Correct. We cannot filter by pid.

Reply:
Thanks @cosmo0920.

Can we add some examples?

Does the max-length include the path to the binary? 15 seems super small...

Reply:
Yup. This seems to be super small and I thought it was wrong.... However, the specification is correct.

The limits is originated from: 
* https://elixir.bootlin.com/linux/latest/source/include/linux/sched.h#L292
* https://elixir.bootlin.com/linux/latest/source/include/linux/sched.h#L1073

And according to man 2 prctl:

> PR_SET_NAME (since Linux 2.6.9)
              Set  the  name  of the calling thread, using the value in the location pointed to by (char *) arg2.  The name can be up to 16 bytes long, including the terminating null byte.  (If the length of
              the string, including the terminating null byte, exceeds 16 bytes, the string is silently truncated.)  This is the same attribute that can be set via pthread_setname_np(3) and  retrieved  using
              pthread_getname_np(3).  The attribute is likewise accessible via /proc/self/task/[tid]/comm, where tid is the name of the calling thread.

Reply:
Can we add these details in a sub-section maybe with some examples?

Reply:
Sure, I'll add them